### ISSUE TITLE
[PRE REVIEW]: AnisoCADO: a python package for analytically generating adaptive optics point spread functions for the Extremely Large Telescope

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2393]
**Submitting author:** @astronomyk (<a href="http://orcid.org/0000-0003-0441-9784">Kieran Leschinski</a>)
**Repository:** <a href="https://github.com/astronomyk/AnisoCADO" target ="_blank">https://github.com/astronomyk/AnisoCADO</a>
**Version:** v0.2
**Editor:** Pending
**Reviewer:** Pending
**Managing EiC:** Kristen Thyng

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**Author instructions**

Thanks for submitting your paper to JOSS @astronomyk. **Currently, there isn't an JOSS editor assigned** to your paper.

The author's suggestion for the handling editor is @harpolea.

@astronomyk if you have any suggestions for potential reviewers then please mention them here in this thread (without tagging them with an @). In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission (please start at the bottom of the list).

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.84  T=0.18 s (134.8 files/s, 24308.5 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          13            645           1289           1167
Markdown                         4             92              0            300
TeX                              1             18              0            221
reStructuredText                 3            176            127            221
DOS Batch                        1              8              1             27
YAML                             1              2              0             15
make                             1              4              6             10
-------------------------------------------------------------------------------
SUM:                            24            945           1423           1961
-------------------------------------------------------------------------------


Statistical information for the repository '2393' was gathered on 2020/06/25.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Kieran                          27          7164           4063          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Kieran                     3101           43.3          9.3               15.58
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02393/joss.02393/10.21105.joss.02393.pdf)

**Comment by astronomyk** [comment]
Potential reviewers:

brechmos
Deech08
cmillion
KshitijAggarwal
devanshkv
zpace
garrettj403
nespinoza
ygrange


-------------------------------------
Kieran Leschinski
Department of Astrophysics
University of Vienna


Am Do., 25. Juni 2020 um 20:05 Uhr schrieb whedon <notifications@github.com

**Comment by arfon** [comment]
:wave: @astronomyk - Thanks for your submission to JOSS. As described in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss) announcing the reopening of JOSS, we're currently working in a "reduced service mode", limiting the number of papers assigned to any individual editor.

Since reopening JOSS earlier in the month we've had a very large number of papers submitted and as such, yours has been put in our backlog that we will be working through over the coming weeks and months.

Thanks in advance for your patience!

**Comment by danielskatz** [comment]
👋 @dfm - would you be able to edit this submission?

**Comment by danielskatz** [comment]
@whedon invite @dfm as editor

**Comment by whedon** [comment]
@dfm has been invited to edit this submission.

**Comment by arfon** [comment]
👋 @astronomyk, thanks for your submission to JOSS. I have a few questions about your submission before we can go out for review:

- This software looks to have been written for a very specific purpose (simulations for the MICADO instrument on the ELT). It looks like there's capability for including a custom atmosphere model, but it's not clear how to use the software in that way?
- Is this software suitable for more general use at other sites/observatories or do you consider this software for just this instrument/observatory combination?

Basically, if the software can be applied to other instrument/observatories then I think it would be good to make it clear this is possible somewhere in the documentation/examples.

**Comment by astronomyk** [comment]
@arfon sorry for the late response. Thanks for the tips regarding extending the documentation.

In answer to your questions:

* there's capability for including a custom atmosphere model,
  
  Indeed there is the ability for this. If the user knows the Cn2 profile of the atmosphere, they can include this directly by setting the `.layerAltitude` and `.Cn2h` attribute. Otherwise, the standard 5 ESO profile (Q1-4, plus Median) are included for convenience. I agree the documentation doesn't explain how to do this though. I will add that to my list of updates for AnisoCADO.

* software suitable for more general use

  the original reason for writing the software was for the MICADO instrument. Hence a lot of the psf_utils functions are related to generating a pupil for the ELT. However essentially any other AO system could be modelled assuming there is a function available to generate the optical transfer function for telescopes pupil. Currently there is no "generic" code for doing arbitary telescope pupils, but we have plans to bring [Poppy](https://github.com/spacetelescope/poppy) into the mix in order to do exactly this. No point in reinventing the wheel.

  So to answer the question of suitability for other instruments: Off the shelf, it is currently only good at simulating ELT PSFs. However if the user has an image of their own telescope pupil, then it can be used to generate PSFs for other observatories. Another point for me to make clear in the documentation.

**Comment by labarba** [comment]
@whedon query scope

**Comment by arfon** [comment]
Thanks for providing this additional information @astronomyk. After consulting with the JOSS editorial team, we've concluded that this package is (currently) out of scope for JOSS.

We'd welcome a resubmission if a major effort was made to make this package more generalized and suitable for other instruments.

**Comment by arfon** [comment]
@whedon reject

**Comment by whedon** [comment]
Paper rejected.
